session.py

```diff
@@ -31,13 +28,25 @@ def __init__(self):
 
 class BidiSessionSignature(Signature):
     """
-    - Start with inbound.active raised: "new request"
-    - Wait for outbound.active to be raised before inputting data
-    - Data flows
-    - Wait until outbound data flushed before deasserting session_active
-    - Early termination (e.g. 400): flush inbound data until inbound.session_active deasserted
-    - Both zero --> reset state
-
+    A bidirectional session-oriented data stream.
+    A pair of SessionSignatures that control inbound and outbound data,
+    and provide a bidirectional handshake for starting a new session.
+
+    The sequence of usage is as follows:
+
+    - At reset, inbound.active and outbound.active are both zero.
+    - A new session starts with inbound.active raised
+    - The session manager wait for outbound.active to be raised before
+      passing data into inbound.data. inbound.data and outbound.data
+      perform flow control as usual
+    - Data flows for a while
+    - At the end of a session, inbound.active OR outbound.active deasserts,
```

Comment:
I might phrase this differently:

- To end the session, both inbound.active and outbound.active will de-assert.
  This can happen in any order, but once a signal has deasserted, it cannot assert
  again until the beginning of the next session.

Reply:
Done in f2985bf